torch/csrc/jit/passes/onnx/shape_type_inference.cpp

```diff
@@ -1153,6 +1375,21 @@ void ComputeConstant(Node* n, int opset_version) {
         only_rank_available = true;
         rank = rank_0 + rank_1 - 1;
       }
+      if (ConstantValueMap::HasShapeValue(n->input(0)->debugName()) &&
+          ConstantValueMap::HasValue(n->input(1)->debugName())) {
+        auto shape_value =
+            ConstantValueMap::GetShapeValue(n->input(0)->debugName()).value();
+        auto idx_value =
+            ConstantValueMap::GetValue(n->input(1)->debugName()).value();
+        if (idx_value.dim() == 0) {
+          auto idx_value_0 = idx_value.item<int64_t>();
+          if (idx_value_0 >= 0) {
+            std::vector<c10::ShapeSymbol> dims = {shape_value.at(idx_value_0)};
```

Comment:
quoting   
```
  // shapeValueMap stores the value of the tensor from a node when this tensor
  // represents a shape.
```

So `shape_value` is the value of input_0(i.e. data) of Gather node. Should we also check that it is a 1d tensor (looks like it is assumed here)?

Reply:
`shapeValueMap` represents the shape. For a Gather input, we check `if (ConstantValueMap::HasShapeValue(n->input(0)->debugName())`, this condition actually means the data input of Gather represent a shape. Then we can apply `shape_value.at(idx_value_0)` here. What kind of check do you refer to?